spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/security/CognitoAuthenticationAutoConfiguration.java

```diff
@@ -0,0 +1,103 @@
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.security.oauth2.resource.servlet.OAuth2ResourceServerAutoConfiguration;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.oauth2.core.DelegatingOAuth2TokenValidator;
+import org.springframework.security.oauth2.core.OAuth2TokenValidator;
+import org.springframework.security.oauth2.jose.jws.SignatureAlgorithm;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.jwt.JwtClaimValidator;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.security.oauth2.jwt.JwtValidators;
+import org.springframework.security.oauth2.jwt.NimbusJwtDecoder;
+
+/**
+ * @author Eddú Meléndez
+ * @since 2.3
+ */
+@Configuration(proxyBeanMethods = false)
+@AutoConfigureBefore(OAuth2ResourceServerAutoConfiguration.class)
+@EnableConfigurationProperties(CognitoAuthenticationProperties.class)
+@ConditionalOnProperty(prefix = "spring.cloud.aws.security.cognito", name = "enabled",
+		matchIfMissing = true)
+@Conditional(CognitoAuthenticationAutoConfiguration.OnUserPoolIdAndRegionPropertiesCondition.class)
+public class CognitoAuthenticationAutoConfiguration {
+
+	private final CognitoAuthenticationProperties properties;
+
+	public CognitoAuthenticationAutoConfiguration(
+			CognitoAuthenticationProperties properties) {
+		this.properties = properties;
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(name = "cognitoJwtDelegatingValidator")
+	public DelegatingOAuth2TokenValidator<Jwt> cognitoJwtDelegatingValidator() {
+		List<OAuth2TokenValidator<Jwt>> validators = new ArrayList<>();
+		validators
+				.add(JwtValidators.createDefaultWithIssuer(this.properties.getIssuer()));
+		validators.add(new JwtClaimValidator<String>("aud",
+				aud -> aud != null && aud.equals(this.properties.getAppClientId())));
+
+		return new DelegatingOAuth2TokenValidator<>(validators);
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public JwtDecoder cognitoJwtDecoder(
+			@Qualifier("cognitoJwtDelegatingValidator") DelegatingOAuth2TokenValidator<Jwt> validator) {
+		NimbusJwtDecoder jwtDecoder = NimbusJwtDecoder
+				.withJwkSetUri(this.properties.getRegistry())
+				.jwsAlgorithm(SignatureAlgorithm.from(this.properties.getAlgorithm()))
+				.build();
+		jwtDecoder.setJwtValidator(validator);
+
+		return jwtDecoder;
+	}
+
+	static class OnUserPoolIdAndRegionPropertiesCondition extends AllNestedConditions {
+
+		OnUserPoolIdAndRegionPropertiesCondition() {
+			super(ConfigurationPhase.PARSE_CONFIGURATION);
+		}
+
+		@ConditionalOnProperty("spring.cloud.aws.security.cognito.userPoolId")
```

Comment:
Properties should not be referenced with camel case. Here: `spring.cloud.aws.security.cognito.user-pool-id`